# Projet 2 : Explorer et comprendre les performances des employés

# Importation des librairies et des data

In [1]:
# Librairies
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import pearsonr
from scipy.stats import spearmanr
from scipy.stats import chi2_contingency
from scipy.stats import fisher_exact
from scipy.stats import ttest_ind
from scipy.stats import f_oneway

In [2]:
# Data 
HR=pd.read_csv('HRDataset_v14.csv')
display(HR)

Employee_Name  EmpID  MarriedID  MaritalStatusID  GenderID  \
0         Adinolfi, Wilson  K  10026          0                0         1   
1    Ait Sidi, Karthikeyan     10084          1                1         1   
2           Akinkuolie, Sarah  10196          1                1         0   
3                Alagbe,Trina  10088          1                1         0   
4            Anderson, Carol   10069          0                2         0   
..                        ...    ...        ...              ...       ...   
306            Woodson, Jason  10135          0                0         1   
307        Ybarra, Catherine   10301          0                0         0   
308          Zamora, Jennifer  10010          0                0         0   
309               Zhou, Julia  10043          0                0         0   
310             Zima, Colleen  10271          0                4         0   

     EmpStatusID  DeptID  PerfScoreID  FromDiversityJobFairID  Salary  ...  \
0              1       5            4                       0   62506  ...   
1              5       3            3                       0  104437  ...   
2              5       5            3                       0   64955  ...   
3              1       5            3                       0   64991  ...   
4              5       5            3                       0   50825  ...   
..           ...     ...          ...                     ...     ...  ...   
306            1       5            3                       0   65893  ...   
307            5       5            1                       0   48513  ...   
308            1       3            4                       0  220450  ...   
309            1       3            3                       0   89292  ...   
310            1       5            3                       0   45046  ...   

        ManagerName  ManagerID  RecruitmentSource PerformanceScore  \
0    Michael Albert       22.0           LinkedIn          Exceeds   
1        Simon Roup        4.0             Indeed      Fully Meets   
2    Kissy Sullivan       20.0           LinkedIn      Fully Meets   
3      Elijiah Gray       16.0             Indeed      Fully Meets   
4    Webster Butler       39.0      Google Search      Fully Meets   
..              ...        ...                ...              ...   
306  Kissy Sullivan       20.0           LinkedIn      Fully Meets   
307  Brannon Miller       12.0      Google Search              PIP   
308      Janet King        2.0  Employee Referral          Exceeds   
309      Simon Roup        4.0  Employee Referral      Fully Meets   
310   David Stanley       14.0           LinkedIn      Fully Meets   

     EngagementSurvey EmpSatisfaction SpecialProjectsCount  \
0                4.60               5                    0   
1                4.96               3                    6   
2                3.02               3                    0   
3                4.84               5                    0   
4                5.00               4                    0   
..                ...             ...                  ...   
306              4.07               4                    0   
307              3.20               2                    0   
308              4.60               5                    6   
309              5.00               3                    5   
310              4.50               5                    0   

    LastPerformanceReview_Date DaysLateLast30 Absences  
0                    1/17/2019              0        1  
1                    2/24/2016              0       17  
2                    5/15/2012              0        3  
3                     1/3/2019              0       15  
4                     2/1/2016              0        2  
..                         ...            ...      ...  
306                  2/28/2019              0       13  
307                   9/2/2015              5        4  
308                  2/21/2019             

In [5]:
# Dimension de la base de donnée
HR.shape

(311, 36)

La base de donnée est composée alors de **311** individus caractérisés par **36** variables.

In [3]:
# Determination des différents types de variables dans notre base de donnéess
HR.dtypes

Employee_Name                  object
EmpID                           int64
MarriedID                       int64
MaritalStatusID                 int64
GenderID                        int64
EmpStatusID                     int64
DeptID                          int64
PerfScoreID                     int64
FromDiversityJobFairID          int64
Salary                          int64
Termd                           int64
PositionID                      int64
Position                       object
State                          object
Zip                             int64
DOB                            object
Sex                            object
MaritalDesc                    object
CitizenDesc                    object
HispanicLatino                 object
RaceDesc                       object
DateofHire                     object
DateofTermination              object
TermReason                     object
EmploymentStatus               object
Department                     object
ManagerName 

In [10]:
# Analyse de l'ensemble des variables de la base de données
HR.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 311 entries, 0 to 310
Data columns (total 36 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Employee_Name               311 non-null    object 
 1   EmpID                       311 non-null    int64  
 2   MarriedID                   311 non-null    int64  
 3   MaritalStatusID             311 non-null    int64  
 4   GenderID                    311 non-null    int64  
 5   EmpStatusID                 311 non-null    int64  
 6   DeptID                      311 non-null    int64  
 7   PerfScoreID                 311 non-null    int64  
 8   FromDiversityJobFairID      311 non-null    int64  
 9   Salary                      311 non-null    int64  
 10  Termd                       311 non-null    int64  
 11  PositionID                  311 non-null    int64  
 12  Position                    311 non-null    object 
 13  State                       311 non

In [12]:
# Combien y a-t-il de valeurs différentes par colonne ? 
HR.nunique()

Employee_Name                 311
EmpID                         311
MarriedID                       2
MaritalStatusID                 5
GenderID                        2
EmpStatusID                     5
DeptID                          6
PerfScoreID                     4
FromDiversityJobFairID          2
Salary                        308
Termd                           2
PositionID                     30
Position                       32
State                          28
Zip                           158
DOB                           307
Sex                             2
MaritalDesc                     5
CitizenDesc                     3
HispanicLatino                  4
RaceDesc                        6
DateofHire                    101
DateofTermination              96
TermReason                     18
EmploymentStatus                3
Department                      6
ManagerName                    21
ManagerID                      23
RecruitmentSource               9
PerformanceSco

## la gestion des valeurs manquantes et incomplètes

In [6]:
# Decompte de valeurs manquantes ou incomplètes par variables
HR.isnull().sum()

Employee_Name                   0
EmpID                           0
MarriedID                       0
MaritalStatusID                 0
GenderID                        0
EmpStatusID                     0
DeptID                          0
PerfScoreID                     0
FromDiversityJobFairID          0
Salary                          0
Termd                           0
PositionID                      0
Position                        0
State                           0
Zip                             0
DOB                             0
Sex                             0
MaritalDesc                     0
CitizenDesc                     0
HispanicLatino                  0
RaceDesc                        0
DateofHire                      0
DateofTermination             207
TermReason                      0
EmploymentStatus                0
Department                      0
ManagerName                     0
ManagerID                       8
RecruitmentSource               0
PerformanceSco

La base des

## La detection des doublons

In [7]:
# Compter les doublons
HR.duplicated().sum()

0

Ce resultat démontre que notre base ne compte pas de doublons.

## La detection des outliers

In [17]:
HR.describe()

EmpID   MarriedID  MaritalStatusID    GenderID  EmpStatusID  \
count    311.000000  311.000000       311.000000  311.000000   311.000000   
mean   10156.000000    0.398714         0.810289    0.434084     2.392283   
std       89.922189    0.490423         0.943239    0.496435     1.794383   
min    10001.000000    0.000000         0.000000    0.000000     1.000000   
25%    10078.500000    0.000000         0.000000    0.000000     1.000000   
50%    10156.000000    0.000000         1.000000    0.000000     1.000000   
75%    10233.500000    1.000000         1.000000    1.000000     5.000000   
max    10311.000000    1.000000         4.000000    1.000000     5.000000   

           DeptID  PerfScoreID  FromDiversityJobFairID         Salary  \
count  311.000000   311.000000              311.000000     311.000000   
mean     4.610932     2.977492                0.093248   69020.684887   
std      1.083487     0.587072                0.291248   25156.636930   
min      1.000000     1.000000                0.000000   45046.000000   
25%      5.000000     3.000000                0.000000   55501.500000   
50%      5.000000     3.000000                0.000000   62810.000000   
75%      5.000000     3.000000                0.000000   72036.000000   
max      6.000000     4.000000                1.000000  250000.000000   

            Termd  PositionID           Zip   ManagerID  EngagementSurvey  \
count  311.000000  311.000000    311.000000  303.000000        311.000000   
mean     0.334405   16.845659   6555.482315   14.570957          4.110000   
std      0.472542    6.223419  16908.396884    8.078306          0.789938   
min      0.000000    1.000000   1013.000000    1.000000          1.120000   
25%      0.000000   18.000000   1901.500000   10.000000          3.690000   
50%      0.000000   19.000000   2132.000000   15.000000          4.280000   
75%      1.000000   20.000000   2355.000000   19.000000          4.700000   
max      1.000000   30.000000  98052.000000   39.000000          5.000000   

       EmpSatisfaction  SpecialProjectsCount  DaysLateLast30    Absences  
count       311.000000            311.000000      311.000000  311.000000  
mean          3.890675              1.218650        0.414791   10.237942  
std           0.909241              2.349421        1.294519    5.852596  
min           1.000000              0.000000        0.000000    1.000000  
25%           3.000000              0.000000        0.000000    5.000000  
50%           4.000000              0.000000        0.000000   10.000000  
75%           5.000000              0.000000        0.000000   15.000000  
max           5.000000              8.000000        6.000000   20.000000

In [19]:
HR.describe(include='object')

Employee_Name                 Position State       DOB  Sex  \
count                   311                      311   311       311  311   
unique                  311                       32    28       307    2   
top     Adinolfi, Wilson  K  Production Technician I    MA  09/09/65    F   
freq                      1                      137   276         2  176   

       MaritalDesc CitizenDesc HispanicLatino RaceDesc DateofHire  \
count          311         311            311      311        311   
unique           5           3              4        6        101   
top         Single  US Citizen             No    White  1/10/2011   
freq           137         295            282      187         14   

       DateofTermination         TermReason EmploymentStatus  \
count                104                311              311   
unique                96                 18                3   
top            8/19/2018  N/A-StillEmployed           Active   
freq                   2                207              207   

               Department     ManagerName RecruitmentSource PerformanceScore  \
count                 311             311               311              311   
unique                  6              21                 9                4   
top     Production         Michael Albert            Indeed      Fully Meets   
freq                  209              22                87              243   

       LastPerformanceReview_Date  
count                         311  
unique                        137  
top                     1/14/2019  
freq                           18

In [ ]:
sns.boxplot(

In [1]:
store = pd.read_csv('superstore.csv')
display(store)

NameError: name 'pd' is not defined